File: Programming Fundamentals with Python/Functions/tribonacci_sequence.py

def tribonacci_sequence(n):
    if n <= 0:
        return []
    elif n == 1:
        return [1]
    elif n == 2:
        return [1, 1]
    elif n == 3:
        return [1, 1, 2]

    sequence = [1, 1, 2]
    for i in range(3, n):
        next_number = sequence[i - 1] + sequence[i - 2] + sequence[i - 3]
        sequence.append(next_number)

    return sequence

def print_tribonacci_sequence(n):
    sequence = tribonacci_sequence(n)
    print(" ".join(str(num) for num in sequence))


n = int(input())
print_tribonacci_sequence(n)

File: Programming Fundamentals with Java/Methods/TribonacciSequence.java
import java.util.Scanner;

public class TribonacciSequence 
{
    private static void printTribonacciSequence(int num) 
    {
        int[] tribonacci = new int[num];

        tribonacci[0] = 1;
        tribonacci[1] = 1;
        tribonacci[2] = 2;

        for (int i = 3; i < num; i++) 
        {
            tribonacci[i] = tribonacci[i - 1] + tribonacci[i - 2] + tribonacci[i - 3];
        }

        for (int i = 0; i < num; i++) 
        {
            System.out.print(tribonacci[i] + " ");
        }
    }

    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int num = input.nextInt();
        printTribonacciSequence(num);
    }
}
